### ISSUE TITLE
Getting nan as KL loss with highly deep networks

### ISSUE THREAD

**Issue by mrivar** [tensorflow/probability#334]
Hi everyone!
I am training a ResNet-34 model with 3D Convolutional bayesian layers. After the first step, regardless of the initial parameters, the KL calculation of the losses (`sum(net.losses) * beta`) goes to nan, being the global loss also nan.

I believe the error is similar to the one in issue #233; that being that the gradient of the KL losses when doing backpropagation is too small - so small that it is below numerical precision.

I am using `tensorflow 1.12.0`, `tensorflow probability 0.5.0` and `python 3.6`
Thanks!

**Comment by axch** [comment]
@mrivar Sorry for not responding earlier.  Is this still an issue for you in TF 1.13 (or the TF 2.0 pre-release) and TFP 0.6 (or nightly)?  If so, could you post a complete (ideally minimal) reproduction?

**Comment by srvasude** [comment]
@mrivar please reopen if this issue is still affecting you. #233 is fixed now, so hopefully your issue should be as well.
